File: stream-state/src/main/java/com/levnovikov/stream_state/AppStateStreamProvider.java
package com.levnovikov.stream_state;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

/**
 * Author: lev.novikov
 * Date: 14/12/17.
 */

public class AppStateStreamProvider {

    @Inject
    public AppStateStreamProvider() { }

    private BehaviorSubject<AppState> appStateSubject = BehaviorSubject.createDefault(AppState.PREBOOKING);
    public Observable<AppState> provideAppStateStream() {
        return appStateSubject.distinctUntilChanged(); //TODO stub
    }
}

File: stream-state/src/main/java/com/levnovikov/stream_state/AppStateStreamProvider.kt
package com.levnovikov.stream_state

import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Inject

/**
 * Author: lev.novikov
 * Date: 14/12/17.
 */

class AppStateStreamProvider @Inject
constructor() {

    private val appStateSubject = BehaviorSubject.createDefault(AppState.PREBOOKING)
    fun provideAppStateStream(): Observable<AppState> {
        return appStateSubject.distinctUntilChanged() //TODO stub
    }
}
